sway-ir/src/parser.rs

```diff
@@ -66,31 +67,23 @@ mod ir_builder {
                         name,
                         args,
                         ret_type,
-                        is_public: true,
+                        is_public,
                         metadata,
                         locals,
                         blocks,
-                        selector
-                    }
-                }
-                / "fn" _ name:id() _ selector:selector_id()? _ "(" _
-                        args:(block_arg() ** comma()) ")" _ "->" _ ret_type:ast_ty()
-                            metadata:comma_metadata_idx()? "{" _
-                        locals:fn_local()*
-                        blocks:block_decl()*
-                    "}" _ {
-                    IrAstFnDecl {
-                        name,
-                        args,
-                        ret_type,
-                        is_public: false,
-                        metadata,
-                        locals,
-                        blocks,
-                        selector
+                        selector,
+                        is_entry
                     }
                 }
 
+            rule is_public() -> bool
```

Comment:
This is probably going away eventually. `pub` is really not relevant in the IR - it doesn't serve any purpose outside of the frontend.